Kudu.Core/Deployment/DeploymentManager.cs

```diff
@@ -284,8 +286,14 @@ public void Delete(string id)
             }
         }
 
-        public async Task RestartMainSiteIfNeeded(ITracer tracer, ILogger logger)
+        public async Task RestartMainSiteIfNeeded(ITracer tracer, ILogger logger, DeploymentInfoBase deploymentInfo)
         {
+            if (deploymentInfo != null && deploymentInfo.Deployer == Constants.OneDeploy)
```

Comment:
Move this after the check if (!_settings.RestartAppOnGitDeploy()) on the next line, so that if restart is disabled, we will skip restarting.